src/cascadia/TerminalSettingsEditor/Profiles_Base.cpp

```diff
@@ -29,31 +29,13 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
     void Profiles_Base::OnNavigatedTo(const NavigationEventArgs& e)
     {
-        auto state{ e.Parameter().as<Editor::ProfilePageNavigationState>() };
-        _Profile = state.Profile();
+        _Profile = e.Parameter().as<Editor::ProfileViewModel>();
 
         // Check the use parent directory box if the starting directory is empty
         if (_Profile.StartingDirectory().empty())
         {
             StartingDirectoryUseParentCheckbox().IsChecked(true);
         }
-
-        _layoutUpdatedRevoker = LayoutUpdated(winrt::auto_revoke, [state, this](auto /*s*/, auto /*e*/) {
-            // This event fires every time the layout changes, but it is always the last one to fire
-            // in any layout change chain. That gives us great flexibility in finding the right point
-            // at which to initialize our renderer (and our terminal).
-            // Any earlier than the last layout update and we may not know the terminal's starting size.
-
-            // Only let this succeed once.
-            _layoutUpdatedRevoker.revoke();
-
-            if (state.FocusDeleteButton())
-            {
-                DeleteButton().Focus(FocusState::Programmatic);
-                state.FocusDeleteButton(false);
-                ProfilesBase_ScrollView().ChangeView(nullptr, ProfilesBase_ScrollView().ScrollableHeight(), nullptr);
-            }
```

Comment:
As far as I can tell, there was no instance where we were setting `FocusDeleteButton` to true, making this obsolete (I'm guessing we didn't need it anymore at some point but didn't remove the handling for it).

Reply:
Tested your branch. We get a regression of #11971. Not sure if the regression exists on `main` but we should definitely keep something like this around.

See #12558 for more details on why this code was added.

Reply:
Turns out, #11971 regressed when we implemented CSVM (the issue exists in main) - that's my bad, I'll make a separate PR to fix it. 

As for this chunk of code, added it back in.